src/registration_and_login/RegistrationForm.js

```diff
@@ -0,0 +1,93 @@
+import React, { useState } from 'react';
+import axios from 'axios';
+import validator from 'validator';
+
+const RegistrationForm = () => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [isValidEmail, setIsValidEmail] = useState(true);
+  const [isValidPassword, setIsValidPassword] = useState(true);
+
+  const handlePasswordChange = (event) => {
+    const newPassword = event.target.value;
+    setPassword(newPassword);
+    setIsValidPassword(validatePassword(newPassword));
+  };
+
+  const validatePassword = (password) => {
+    // Custom password validation rules
+    // Example: At least 8 characters and contain a lowercase letter, an uppercase letter, and a number
+    return (
+      validator.isLength(password, { min: 8 }) &&
+      /[a-z]/.test(password) &&
+      /[A-Z]/.test(password) &&
+      /\d/.test(password)
+    );
+  };
+
+  const handleEmailChange = (event) => {
+    const newEmail = event.target.value;
+    setEmail(newEmail);
+    setIsValidEmail(validator.isEmail(newEmail));
+  };
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+
+    if (isValidEmail) {
+      // Perform further actions or submit the form
+      console.log('Valid email:', email);
+    } else {
+      console.log('Invalid email:', email);
+    }
+
+    if (isValidPassword) {
+      // Perform further actions or submit the form
+      console.log('Valid password:', password);
+    } else {
+      console.log('Invalid password:', password);
+    }
+    
+
+    try {
+      console.log(process.env.REACT_APP_API_ROOT_VAR);

```

Comment:
A common practice is not to keep logs in production code.

Reply:
Should I remove all logs? error logs too?

Reply:
This will run on your user's browser so only they can see the logs, which is not very useful for them, nor for you. But this try clause is expecting a possible failure of the axios request and that should be handled in the UI — the user should know that something went wrong and you should give her an option to retry. But most users won't have the devtools open to see the log in the console.